src/Storages/MergeTree/ReplicatedMergeTreeQueue.h

```diff
@@ -163,7 +163,7 @@ class ReplicatedMergeTreeQueue
     /// A subscriber callback is called when an entry queue is deleted
     mutable std::mutex subscribers_mutex;
 
-    using SubscriberCallBack = std::function<void(size_t /* queue_size */)>;
+    using SubscriberCallBack = std::function<void(size_t /* queue_size */, const String & /* removed_log_entry_id*/)>;
```

Comment:
maybe make `removed_log_entry_id` std::optional<std::string> to avoid passing empty string if queue empty.

Reply:
ya good point. addressed it.